### PR TITLE
Feat/refactoring retry

### PR DESCRIPTION
**Pull request by MaximeJEUSS**
- Make a Client without timeout and remove the retry when the request is a create session

### CODE REVIEW

**Comment by theredfish** [comment]
@MaximeJEUSS  if the pull request is done you can remove the draft and ask for a review.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 target
+Makefile
```
Comment (theredfish) [inline .gitignore:2]: Remove this and do not commit your own Makefile. This information isn't relevant in the context of this project. Not everybody will use a Makefile in local.

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "soda-test-service"
-version = "0.3.0"
+version = "0.3.3"
```
Comment (theredfish) [inline Cargo.toml:3]:
```suggestion
version = "0.3.1"
```

Not sure why you are patching 3 versions in one... If you already pushed a private version of the test-service with 2 other patches you will need to yank them and publish the `0.3.1` patch again.

File: README.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ Copy the following snippet to add the test service in your docker-compose file a
 
 ```
 test-service:
-  image: soda/test-service:0.3.0
+  image: soda/test-service:0.3.3
```
Comment (theredfish) [inline README.md:36]: same for the patch version

File: src/inspector.rs [hunk 4]
```diff
@@ -42,22 +42,27 @@ impl fmt::Display for DeleteEvent {
...
+        info!("{}, Request Id : {:?}", capture_create_event(body).await, id);
     } else if method == "POST" && !is_a_new_session(&path) {
         if let Some(url_event) = capture_url_event(path, body) {
-            info!("{}", url_event);
+            info!("{}, Request Id : {:?}", url_event, id);
         }
     }
 }
 
+pub fn is_a_new_create_session(method: Method, path: &str) -> bool {
```
Comment (theredfish) [inline src/inspector.rs:62]: Is there a case where the path can be exactly equals to `/wd/hub/session` while the HTTP method is different from `POST`? If not the case, `new_session()` already check this as explained by the documentation : 

```rust
/// Split the path to determine if it's a new session
/// (the path doesn't contain the session's id) or if it's
/// an existing session (the path contains the session's id).
/// If the head and the tail of the path are empty,
/// it's a new session event that we want to capture.
```

File: src/reverse_proxy.rs [hunk 5]
```diff
@@ -108,13 +108,30 @@ pub async fn forward(req: Request<Body>, state: AppState) -> Result<Response<Bod
...
+                    "Request Id : {:?} Try number {} in error for response unwrap : {}",
+                    request_id, tries, err
+                )
+            });
+        match res {
+            Err(e) if tries <= 3 && !is_a_new_create_session => {
+                tries += 1;
+                log::error!("{}", e);
+            }
+            res => return res,
```
Comment (theredfish) [inline src/reverse_proxy.rs:134]: Maybe there is another way to handle the loop result : https://doc.rust-lang.org/stable/rust-by-example/flow_control/loop/return.html

File: src/reverse_proxy.rs [hunk 6]
```diff
@@ -108,13 +108,30 @@ pub async fn forward(req: Request<Body>, state: AppState) -> Result<Response<Bod
 pub async fn send_request(
```
Comment (theredfish) [inline src/reverse_proxy.rs:108]: Concerning the retry, it's currently hard to follow what you written. I suggest you to :

- Put the retry logic in another module
- Or use an existing library :) https://docs.rs/retry/1.1.0/retry/

File: src/reverse_proxy.rs [hunk 7]
```diff
@@ -108,13 +108,30 @@ pub async fn forward(req: Request<Body>, state: AppState) -> Result<Response<Bod
...
     client: Client,
     method: Method,
+    request_id: Uuid,
```
Comment (theredfish) [inline src/reverse_proxy.rs:111]: This additional information concerning the request makes me thing it could be better to use a structure instead of these arguments. Maybe something like :

```rust
struct CapturedRequest {
  method: Method,
  request_id: Uuid,
  body_bytes: Bytes, // maybe this can be optional so we can reuse the structure for requests with an empty body
  url: Url
}
```

File: src/reverse_proxy.rs [hunk 8]
```diff
@@ -26,7 +31,7 @@ pub async fn forward(req: Request<Body>, state: AppState) -> Result<Response<Bod
         .map(|x| x.to_string())
         .unwrap_or_else(|| "".to_string());
 
-    info!("{} {}", method, path);
+    info!("{:?} {} {}", request_id, method, path);
```
Comment (theredfish) [inline src/reverse_proxy.rs:34]: You can avoid the debug call (`{:?}`) by calling `request_id.to_string()` I think.

File: docker-compose.yml [hunk 9]
```diff
@@ -34,7 +34,7 @@ services:
       - local-network
 
   test-service:
-    image: soda/test-service:0.3.0
+    image: soda/test-service:0.3.3
```
Comment (theredfish) [inline docker-compose.yml:37]: same for the patch version

**Review by theredfish — CHANGES_REQUESTED** [review]

File: src/inspector.rs (hunk 4 above)
Comment (MaximeJEUSS) [reply src/inspector.rs:62]: Indead

File: src/reverse_proxy.rs (hunk 6 above)
Comment (MaximeJEUSS) [reply src/reverse_proxy.rs:108]: We tried an existing library but there was no good ones with the use of async/await functions.
